webhook.rst

```diff
@@ -0,0 +1,133 @@
+
+Usage in combination with the Mailer component
+----------------------------------------------
+
+When using a third-party mailer, you can use the webhook component to receive
+webhook calls from the third-party mailer.
+
+In this example Mailgun is used with ``'mailer_mailgun'`` as webhook type.
+Any type name can be used as long as it's unique. Make sure to use it in the
+routing configuration, the webhook URL and the Remote Event Consumer.
+
+Install the third party mailer as described in the documentation of the mailer
+component.
+
+The webhook component routing needs to be defined:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/packages/framework.yaml
+        framework:
+            webhook:
+                routing:
+                    mailer_mailgun:
+                        service: 'mailer.webhook.request_parser.mailgun'
+                        secret: '%env(MAILER_MAILGUN_SECRET)%'
+
+    .. code-block:: xml
+
+        <!-- config/packages/framework.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xmlns:framework="http://symfony.com/schema/dic/symfony"
+                   xsi:schemaLocation="http://symfony.com/schema/dic/services
+                        https://symfony.com/schema/dic/services/services-1.0.xsd
+                        http://symfony.com/schema/dic/symfony https://symfony.com/schema/dic/symfony/symfony-1.0.xsd">
+            <framework:config>
+                <framework:webhook enabled="true">
+                    <framework:routing type="mailer_mailgun">
+                        <framework:service>mailer.webhook.request_parser.mailgun</framework:service>
+                        <framework:secret>%env(MAILER_MAILGUN_SECRET)%</framework:secret>
+                    </framework:routing>
+                </framework:webhook>
+            </framework:config>
+        </container>
+
+    .. code-block:: php
+
+        // config/packages/framework.php
+        use App\Webhook\MailerWebhookParser;
+        use Symfony\Config\FrameworkConfig;
+        return static function (FrameworkConfig $frameworkConfig): void {
+            $webhookConfig = $frameworkConfig->webhook();
+            $webhookConfig
+                ->routing('mailer_mailgun')
+                ->service('mailer.webhook.request_parser.mailgun')
+                ->secret('%env(MAILER_MAILGUN_SECRET)%')
+            ;
+        };
+
+Currently, the following third-party mailer services support webhooks:
+
+=============== ==========================================
+Mailer service  Parser service name
+=============== ==========================================
+Mailgun         ``mailer.webhook.request_parser.mailgun``
+Postmark        ``mailer.webhook.request_parser.postmark``
+Sendgrid        ``mailer.webhook.request_parser.sendgrid``
```

Comment:
IIIC Sendgrid was added in 6.4

Reply:
If this is true we should only list 6.3 ones. And add Sendgrid in 6.4 after merge with a versionadded directive

Reply:
You're right. 6.3 only has Mailgun and Postmark.
6.4 added Sendgrid.

I removed Sendgrid for now. I can add it in a separate PR for 6.4.